buddy_alloc.h

```diff
@@ -1231,27 +1368,30 @@ static void buddy_tree_grow(struct buddy_tree *t, uint8_t desired_order) {
             next_pos = buddy_tree_parent(next_pos);
         }
         /* Advance the order and refresh the root */
-        t->order += 1u;
+        t->order = (uint8_t)(t->order + 1u);
```

Comment:
This is a warning of implicit convert to uint8_t. By the way, we assume the actual usecase the order won't overflow?

Reply:
Correct, the tree order is not expected to overflow. A 255-order tree would be able to manage a huge amount of memory with a reasonable block size - more than what's physically available nowadays. Say you have 32 bytes as the smallest block size. That's a 32 * (2^255) arena.